### DISCUSSION TITLE
Setup environment variable with export command

### DISCUSSION THREAD

**Discussion by dschinkel** [community/community#26013] [Actions]
<p>I am not sure if this is even possible is it?  If it is, how do I change this to work?</p>
<pre><code class="lang-auto">- run: export CR_PAT=GITHUB_CONTAINER_REGISTRY_PAT
- run: echo ${{ env.CR_PAT }}
</code></pre>
<p>this didn’t seem to be valid either:</p>
<pre><code class="lang-auto">- run: export CR_PAT=GITHUB_CONTAINER_REGISTRY_PAT
- run: echo ${CR_PAT}
</code></pre>
<p>Error: <code>Error: Cannot perform an interactive login from a non TTY device</code></p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The problem is that each <code>run</code> step gets its own shell instance, so the <code>export</code> doesn’t carry over. To set an environment variable for the rest of the job you need to <a href="https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable">use the <code>set-env</code> workflow command</a>:</p>
<pre><code class="lang-yaml">- run: echo "::set-env name=CR_PAT::GITHUB_CONTAINER_REGISTRY_PAT"
</code></pre>
<p>After that either of the references should work. Or, if you need <code>CR_PAT</code> only in that one step you could <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsenv">use an <code>env</code> property for the step</a>:</p>
<pre><code class="lang-yaml">- run: echo ${CR_PAT}
  env:
    CR_PAT: GITHUB_CONTAINER_REGISTRY_PAT
    
</code></pre>
